lib/collection/src/shards/local_shard.rs

```diff
@@ -557,11 +559,14 @@ impl LocalShard {
             rx.await?;
         }
 
+        let temp_path = temp_path.join(format!("{}.snapshot", Uuid::new_v4()));
+        create_dir_all(&temp_path).await?;
```

Comment:
nitpick:

Here we are conflating a bit the temp directory and the actual snapshot file.

I find a bit exotic to do `create_dir_all` on the full path with the snapshot file extension.

I'd prefer for clarity to first create the actual `temp_dir` directory and then create the full path object for the snapshot creation.

Reply:
The `temp_path` (with extension) is a directory, not the final snapshot archive file.

Reply:
Thanks, I was really confused :sweat_smile: 

I like your proposal to make it more conventional!